lib/okta_asa.rb

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+
+require "net/https"
+require "uri"
+require "json"
+
+def okta_asa_delete(team_name, key_id, key_secret, hostname)
+
+  okta_url = "https://app.scaleft.com/v1/teams"
+
+  # Get auth token
+  token_uri = URI.parse("#{okta_url}/#{team_name}/service_token")
+
+  header = {"Content-Type": "application/json"}
+
+  http = Net::HTTP.new(token_uri.host, token_uri.port)
+  http.use_ssl = true
+  request = Net::HTTP::Post.new(token_uri.request_uri, header)
+  request.body = {"key_id": key_id,
+                  "key_secret": key_secret}.to_json
+
+  response = http.request(request)
+
+  if response.code != '200'
+    puts "ERROR: Unable to authenticate to Okta. API Response:"
+    puts response.body
+
+    return 1
+  else
+    bearer_token = JSON.parse(response.body)["bearer_token"]
+  end
+
+  # Find the system
+  system_uri = URI.parse("#{okta_url}/#{team_name}/servers?hostname=#{hostname}")
+
+  header = {"Authorization": "Bearer #{bearer_token}"}
+
+  http = Net::HTTP.new(system_uri.host, system_uri.port)
+  http.use_ssl = true
+  request = Net::HTTP::Get.new(system_uri.request_uri, header)
+
+  response = http.request(request)
+
+  if response.code != '200'
+    puts "ERROR: Unable to find server '#{hostname}' from Okta ASA. API Response:"
+    puts response.body
+
+    return 1
```

Comment:
I'm typically not a fan of multiple returns in a single method, but ignoring that should we make the exit codes unique? Maybe make the first one something like 100, this one 101, then let the `delete_errors` be their own counter like they currently are?

Reply:
I can't think of a scenario (especially in our use-case with rmvm) where the code calling this would behave differently based on where this function fails.